mne/time_frequency/psd.py

```diff
@@ -112,6 +114,202 @@ def _check_nfft(n, n_fft, n_overlap):
     return n_fft, n_overlap
 
 
+def _check_psd_data(inst, tmin, tmax, picks, proj):
+    """Helper to do checks on PSD data / pull arrays from inst"""
+    from ..io.base import _BaseRaw
+    from ..epochs import _BaseEpochs
+    from ..evoked import Evoked
+    if not isinstance(inst, (_BaseEpochs, _BaseRaw, Evoked)):
+        raise ValueError('epochs must be an instance of Epochs, Raw, or'
+                         'Evoked. Got type {0}'.format(type(inst)))
+
+    time_mask = _time_mask(inst.times, tmin, tmax)
+    if picks is None:
+        picks = pick_types(inst.info, meg=True, eeg=True, ref_meg=False,
+                           exclude='bads')
+    if proj:
+        inst = inst.apply_proj()
+
+    sfreq = inst.info['sfreq']
+    if isinstance(inst, _BaseRaw):
+        start, stop = np.where(time_mask)[0][[0, -1]]
+        data, times = inst[picks, start:(stop + 1)]
+    elif isinstance(inst, _BaseEpochs):
+        data = inst.get_data()[:, picks][..., time_mask]
+    elif isinstance(inst, Evoked):
+        data = inst.data[picks][..., time_mask]
+
+    return data, sfreq
+
+
+@verbose
```

Comment:
ok cool - those changes now reflected in the latest commit

On Wed, Dec 30, 2015 at 2:46 PM, Eric Larson <notifications@github.com>
wrote:

> In mne/time_frequency/psd.py
> <https://github.com/mne-tools/mne-python/pull/2710#discussion_r48630303>:
>
> > +    if proj:
> > +        inst = inst.apply_proj()
> > +
> > +    sfreq = inst.info['sfreq']
> > +    if isinstance(inst, _BaseRaw):
> > +        start, stop = np.where(time_mask)[0][[0, -1]]
> > +        data, times = inst[picks, start:(stop + 1)]
> > +    elif isinstance(inst, _BaseEpochs):
> > +        data = inst.get_data()[:, picks][..., time_mask]
> > +    elif isinstance(inst, Evoked):
> > +        data = inst.data[picks][..., time_mask]
> > +
> > +    return data, sfreq
> > +
> > +
> > +@verbose
>
> Yeah. Sometimes we want them for private functions, too, but this is not
> one of those cases.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2710/files#r48630303>.
>